### PR TITLE
Fix issue 869

### PR DESCRIPTION
**Pull request by danascheider**
Here's a fix for issue #869, including passing Cucumber and RSpec tests. I ran all the tests and everything passes, but let me know if you see anything that needs fixing.

### CODE REVIEW

**Comment by danascheider** [comment]
I'm looking into how to fix the JRuby failure my additions caused. I don't have experience working with JRuby, so if anybody knows off the top of their head what this null pointer error is about, please let me know so I can fix it faster.

**Comment by tooky** [comment]
Hey @danascheider - thanks for taking a look at that.

I think the JRuby exception may be a different exception that is being thrown by gherkin when it tries unsuccessfully to initialize a new language.

I think the actual cause of the problem is this line here:

https://github.com/cucumber/gherkin/blob/master/lib/gherkin/i18n.rb#L92

It's checking for a `nil?` `@iso_code`, but I'm not sure that will ever be nil. I suspect it should raise "Language not supported" when `@keywords` is nil.

I don't know if it will be possible to release a fix for that in gherkin - I'm not 100% sure there is a plan to release any more.

cc @aslakhellesoy

**Comment by danascheider** [comment]
Thanks, @tooky, I was able to make the builds pass by checking the validity of the language code before passing it along to Gherkin, and then give the table of allowed languages if the given code is invalid. That oughta fix it.

File: features/lib/step_definitions/language_steps.rb [hunk 1]
```diff
@@ -6,3 +6,7 @@
     expect(all_output.force_encoding('utf-8')).to include(language)
   end
 end
+
+Then(/^the output includes the message "(.*)"$/) do |message|
```
Comment (mattwynne) [inline features/lib/step_definitions/language_steps.rb]: This is a generic step that doesn't belong in 'language steps' - better to put it in `aruba_steps.rb` with the similar steps.

File: features/docs/writing_support_code/after_hooks.feature [hunk 2]
```diff
@@ -11,7 +11,7 @@ Feature: After Hooks
   Background:
     Given the standard step definitions
 
-  Scenario Outline: Retreive the status of a scenario as a symbol
+  Scenario Outline: Retrieve the status of a scenario as a symbol
```
Comment (mattwynne) [inline features/docs/writing_support_code/after_hooks.feature]: Please try and avoid doing this as part of an unrelated PR - better to put these kinds of changes into individual PRs, however small. :)

**Comment by mattwynne** [comment]
Thanks @danascheider! I'll tidy up those little lose ends now it's merged into master.

File: features/docs/writing_support_code/after_hooks.feature (hunk 2 above)
Comment (danascheider) [inline features/docs/writing_support_code/after_hooks.feature]: Will do.

Comment (mattwynne) [inline features/docs/writing_support_code/after_hooks.feature]: Thanks. I know how tempting it is to fix little things like this "while you're there" (and I very much appreciate you doing it!) but it makes it much easier for us to read / review a PR if it's focussed on just one change.

**Pull request merged**